pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -618,3 +617,32 @@ func (s *NSXServiceAccountService) UpdateProxyEndpointsIfNeeded(ctx context.Cont
 	}
 	return nil
 }
+
+func (s *NSXServiceAccountService) CleanupBeforeVPCDeletion(ctx context.Context) error {
+	ccpList := s.ClusterControlPlaneStore.List()
+	if len(ccpList) == 0 {
+		log.Info("No cluster control plane found while cleanup")
+		return nil
+	}
+	for i := range ccpList {
+		ccp := ccpList[i].(*model.ClusterControlPlane)
+		err := s.DeleteClusterControlPlane(ctx, *ccp.Id)
+		if err != nil {
+			log.Error(err, "Cleanup failed to delete cluster control plane", "ClusterControlPlane", ccp.Id)
+			return err
```

Comment:
Can you check other cleanup functions, whether they will continue the deletion or abort the deletion when there is an error? I'm thinking of:

```
var errs []error
for i := range ccpList {
  err := s.DeleteClusterControlPlane(ctx, *ccp.Id)
  if err != nil {
    errs = append(errs, err)
    continue
  }
  ...
}
return errors.Join(errs...)
```

If no other resource has this cleanup pattern, we can abort early instead of continuing.

Reply:
I couldn't find this cleanup pattern.

Reply:
But the parent caller will retry by calling `CleanupBeforeVPCDeletion` method again incase if timeout error is returned by `CleanupBeforeVPCDeletion`.
https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/clean/clean_service.go#L78

Reply:
added, thanks!